src/blueprint/blueprint.rs

```diff
@@ -219,4 +221,53 @@ impl Blueprint {
     // generation of schema cannot fail.
     schema.finish().unwrap()
   }
+
+  /// Generate a dumb schema useful for validation.
+  pub fn to_validation_schema(&self) -> Schema {
+    create(self, Some(SchemaModifiers { no_resolver: true }))
+      .validation_mode(ValidationMode::Strict)
+      .disable_introspection()
+      .finish()
+      .unwrap()
+  }
+
+  pub async fn validate_operation(schema: &Schema, query: &str) -> Vec<Cause<String>> {
+    schema
+      .execute(query)
+      .await
+      .errors
+      .iter()
+      .map(|err| {
+        Cause::new(err.message.clone()).trace(VecDeque::from_iter(
+          err
+            .locations
+            .iter()
+            .map(|loc| format!("line {} column {}", loc.line, loc.column)),
+        ))
+      })
+      .collect()
+  }
+
+  pub async fn validate_operations(&self, operations: Vec<String>) -> Valid<(), String> {
+    let schema = self.to_validation_schema();
+    let mut execution = vec![];
+
+    for op in operations.iter() {
+      match tokio::fs::read_to_string(op).await {
+        Ok(operation) => {
+          let causes = Blueprint::validate_operation(&schema, &operation).await;
+          execution.push((op.to_string(), causes));
+        }
+        Err(_) => execution.push((
+          op.to_string(),
+          vec![Cause::new(format!("Cannot read file operation file {}", op))],
+        )),
+      }
+    }
```

Comment:
we should use `operations.iter().map()` rather than using for and `.push` values into execution.

Reply:
I can use `futures::stream` for that, but that will require adding a new dependency. So I think, this is ok.

Reply:
probably, using `futures_util::future::join_all` will be enough and we have that dependency.

Overall, we want to execute all the checks in parallel and not sequentially since it is expected to finish faster